extractor/src/test/java/org/schabi/newpipe/NewPipeTestRunner.java

```diff
@@ -0,0 +1,308 @@
+package org.schabi.newpipe;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.Description;
+import org.junit.runner.Runner;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.TestTimedOutException;
+import org.schabi.newpipe.extractor.exceptions.ReCaptchaException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class NewPipeTestRunner extends Runner {
```

Comment:
Refer to [my comment](https://github.com/TeamNewPipe/NewPipeExtractor/pull/180#issuecomment-546625307) about the superclass. Some reasons are:
- Individual tests can't be run anymore.
- Hard-to-track bugs may rise from the custom implementation, as it isn't tested.